src/front/app/components/Challenges/index.js

```diff
@@ -3,10 +3,32 @@ import ChallengeList from './ChallengeList';
 import Menu from './Menu';
 import Navigation from './../Navigation';
 
-export default () => (
-    <div>
-        <Menu />
-        <ChallengeList />
-        <Navigation />
-    </div>
-);
+class Challenges extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            isMenuActive: false
+        };
+        this.onScroll = (e) => {
```

Comment:
Methods should be defined on the class body level and only bound in the constructor (this.onScroll = this.onScroll.bind(this))